client/src/components/backlog/PaginatedBacklog.jsx

```diff
@@ -0,0 +1,48 @@
+import React, { useState } from "react";
+import { useQuery } from "@tanstack/react-query";
+import { fetchTasks } from "../../api/taskApi";
+import Backlog from "./Backlog";
+import Pagination from "./Pagination";
+
+const PaginatedBacklog = () => {
+  const [currentPage, setCurrentPage] = useState(1);
+  const pageSize = 10;
+
+  const { data, isLoading, error } = useQuery({
+    queryKey: ["tasks", currentPage],
+    queryFn: () => fetchTasks(currentPage, pageSize),
+  });
+
+  const handlePageChanged = (pageNumber) => {
+    setCurrentPage(pageNumber);
+  };
+
+  const apiPageCount = data?.meta?.pagination?.pageCount || 1;
+  const totalPages = Math.max(apiPageCount, 3);
```

Comment:
Is there a reason for having at least 3 pages? Seems a bit weird to show 3 if there are only 1 or 2.